base/cvd/cuttlefish/host/commands/defaults/defaults.cpp

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (C) 2025 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <string.h>
+#include <unistd.h>
+
+#include <string_view>
+
+#include <android-base/logging.h>
+#include <android-base/macros.h>
+#include "absl/cleanup/cleanup.h"
+#include "absl/flags/flag.h"
+#include "absl/flags/parse.h"
+#include "absl/strings/str_format.h"
+
+#include "cuttlefish/common/libs/key_equals_value/key_equals_value.h"
+#include "cuttlefish/host/libs/web/http_client/curl_http_client.h"
+#include "cuttlefish/host/libs/web/http_client/http_string.h"
+
+ABSL_FLAG(std::string, filename, "/usr/lib/cuttlefish-common/etc/cf_defaults", "Output filename.");
```

Comment:
> We introduce a new defaults binary that runs in GCE instances, which will eventually be deployed with the cuttlefish-integration package, and runs at boot.

It's not clear to me why the flag defaults state needs to be reflected in the filesystem. If there is GCE metadata that affects the flag defaults, is it possible to defer reading that GCE metadata into launch time rather than immediately putting it in the filesystem at GCE instance boot time? The logic for precedence between a preexisting file on the filesystem versus different GCE metadata can exist there.

Reply:
Going through the filesystem has a few risks:

- Needs access permissions to write to a specific file path
- File could be left over from a previous invocation if an instance is rebooted

Another way to satisfy the "GCE metadata is only read from `cuttlefish-integration`" constraint is to have this new executable write data to `stdout` rather than save it to a file. Executables from `cuttlefish-base` can test for presence of the `defaults` executable, and when it exists invoke it and parse the output.